3DCalendar/CalendarView.swift

```diff
@@ -0,0 +1,166 @@
+        myCollectionView.backgroundColor = .clear
+        myCollectionView.allowsMultipleSelection = false
+        return myCollectionView
+        
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        initializeView()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func initializeView() {
+        currentMonthIndex = Calendar.current.component(.month, from: Date())
+        currentYear = Calendar.current.component(.year, from: Date())
+        todaysDate = Calendar.current.component(.day, from: Date())
+        //firstWeekDayOfMonth = getFirstWeekDay()
+        
+        //leap years
+        if currentMonthIndex == 2 && currentYear % 4 == 0 {
+            numDaysPerMonth[currentMonthIndex-1] = 29
+        }
+        
+        presentMonthIndex = currentMonthIndex
+        presentYear = currentYear
+        
+        setUpViews()
+        
+        myCollectionView.delegate = self
+        myCollectionView.dataSource = self
+        myCollectionView.register(dateCell.self, forCellWithReuseIdentifier: "Cell")
+        
+    }
+    /*
+    // Only override draw() if you perform custom drawing.
+    // An empty implementation adversely affects performance during animation.
+    override func draw(_ rect: CGRect) {
+        // Drawing code
+    }
+    */
+   
+    func getFirstWeekDay() -> Int {
+        let day = ("\(currentYear)-\(currentMonthIndex)-01".date?.firstDayOfTheMonth.weekday)!
+        //return day == 7 ? 1 : day
+        return day
+    }
+    
+    func setUpViews() {
+        addSubview(myCollectionView)
+        myCollectionView.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        myCollectionView.leftAnchor.constraint(equalTo: leftAnchor).isActive = true
+        myCollectionView.rightAnchor.constraint(equalTo: rightAnchor).isActive = true
+        myCollectionView.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return numDaysPerMonth[currentMonthIndex-1] + firstWeekDayOfMonth - 1
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath) as! dateCell
+        cell.backgroundColor=UIColor.clear
+        if indexPath.item <= firstWeekDayOfMonth - 2 {
+            cell.isHidden = true
+        } else {
+            let calcDate = indexPath.row-firstWeekDayOfMonth+2
+            cell.isHidden = false
+            cell.label.text="\(calcDate)"
+            if calcDate < todaysDate && currentYear == presentYear && currentMonthIndex == presentMonthIndex {
+                cell.isUserInteractionEnabled=false
```

Comment:
We actually don't need these, the user won't be interacting with the collection view